README.md

```diff
@@ -5,4 +5,123 @@ Leto-Modelizer API is a Python-based API designed to generate infrastructure cod
\ No newline at end of file
+Built on top of Python's robust web framework, the Leto-Modelizer API offers a flexible and extensible architecture for integrating various AI models. Whether you prefer using pre-trained models or training your own custom models, our API seamlessly handles the code generation process. By leveraging the power of AI and Leto-Modelizer, our API simplifies the task of infrastructure code generation, saving you time.
+
+
+## Setup
+
+Currently this project use Ollama in order to generate code.
+So you need to install Ollama on your local: 
+
+```sh 
+curl -fsSL https://ollama.com/install.sh | sh 
+```
+
+Then you have to pull a model from Ollama:
+
+```sh 
+ollama pull mistral 
+```
+
+### Testing Ollama
+
+In order to test Ollama, you can run your model using the following command:
+
+```sh
+ollama run mistral
+```
+
+Then you can just ask it a question !
+
+**NOTE**: You may need to install NVIDIA Container Toolkit to run Ollama.
+cf: https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html
+
+
+## How to launch the API using uvicorn
+
+First install pipenv and create a virtual environment with the needed packages:
+
+* pipenv
+
+    Install pipenv:  ``` pip install pipenv ```
+
+    Create virtual environment and install dependencies from your repo: ``` pipenv install ```
+
+    Install dev dependencies: ``` pipenv install --dev ```
+
+    Activate virtual environment: ``` pipenv shell ```
+
+    To deactivate your virtual environment: ``` deactivate ```
+
+Then launch the API using uvicorn (from the root folder):
+
+```sh
+uvicorn src.main:app --reload
+```
+
+Once it is runing, you can request it on this url: ```http://localhost:8000/```
+
+## How to launch the API using docker
+
+You need to run the following commands to launch the API.
+
+First build the docker image using the following command:
+
+```sh
+docker build -t leto-modelizer-ia-api .
+```
+
+Then run the image:
+
+```sh
+docker run -p 8000:8000 --net=host leto-modelizer-ia-api
+```
+
+Once your docker is runing, you can request it on this url: ```http://localhost:8000/```
+
+## Configuration
+
+To configure the API, you need to edit the `configuration/configuration.json` file (default location).
+Or you can set the environment variables `API_CONFIGURATION` and put the `configuration.json` file in that path`.
+
+Currently the configuration has the following settings:
+
+| Setting            | Description                                                                                     |
+|--------------------|-------------------------------------------------------------------------------------------------|
+| plugin_preferences | A dictionary containing the plugin preferences, which are the AI models to use for what plugin. |
+| ollama             | A dictionary containing the ollama configuration.                                               |
```

Comment:
Rework on this, because ollama is depending on what we setup in pluginPrferenecs.
And explain you can manage all model you want, add a section in contributing to explain how you can add an new model and make a link of contributing section here, like `see how to use another model`

Explain default implemented models are: ollama, here is the configuration example and here is the configruation of ollama.

Reply:
I already explained the configuration for Ollama we have just after ...
But the other section is needed missing in the contributing, i will add it

Reply:
done